simtools/corsika/corsika_config.py

```diff
@@ -0,0 +1,522 @@
+            file.write('\n')
+            file.write('IACT setenv PRMNAME {}\n'.format(self.primary))
+            file.write('IACT setenv ZA {}\n'.format(int(self._userParameters['THETAP'][0])))
+            file.write('IACT setenv AZM {}\n'.format(int(self._userParameters['PHIP'][0])))
+
+            file.write('\n* [ SEEDS ]\n')
+            self._writeSeeds(file)
+
+            file.write('\n* [ TELESCOPES ]\n')
+            telescopeListText = self.layout.getCorsikaInputList()
+            file.write(telescopeListText)
+
+            file.write('\n* [ INTERACTION FLAGS ]\n')
+            textInteractionFlags = _getTextSingleLine(self._corsikaParameters['INTERACTION_FLAGS'])
+            file.write(textInteractionFlags)
+
+            file.write('\n* [ CHERENKOV EMISSION PARAMETERS ]\n')
+            textCherenkov = _getTextSingleLine(
+                self._corsikaParameters['CHERENKOV_EMISSION_PARAMETERS']
+            )
+            file.write(textCherenkov)
+
+            file.write('\n* [ DEBUGGING OUTPUT PARAMETERS ]\n')
+            textDebugging = _getTextSingleLine(
+                self._corsikaParameters['DEBUGGING_OUTPUT_PARAMETERS']
+            )
+            file.write(textDebugging)
+
+            file.write('\n* [ OUTUPUT FILE ]\n')
+            file.write('TELFIL {}\n'.format(self._outputGenericFileName))
+
+            file.write('\n* [ IACT TUNING PARAMETERS ]\n')
+            textIact = _getTextMultipleLines(self._corsikaParameters['IACT_TUNING_PARAMETERS'])
+            file.write(textIact)
+
+            file.write('\nEXIT')
+
+        self._isFileUpdated = True
+    # End of exportInputFile
+
+    def _setOutputFileAndDirectory(self):
+        configFileName = names.corsikaConfigFileName(
+            arrayName=self.layoutName,
+            site=self.site,
+            zenith=self._userParameters['THETAP'],
+            viewCone=self._userParameters['VIEWCONE'],
+            label=self.label
+        )
+        fileDirectory = io.getCorsikaOutputDirectory(self._filesLocation, self.label)
+        fileDirectory.mkdir(parents=True, exist_ok=True)
+        self._logger.info('Creating directory {}, if needed.'.format(fileDirectory))
+        self._configFilePath = fileDirectory.joinpath(configFileName)
+
+        self._outputGenericFileName = names.corsikaOutputGenericFileName(
+            arrayName=self.layoutName,
+            site=self.site,
+            label=self.label
+        )
+    # End of setOutputFileAndDirectory
+
+    def _writeSeeds(self, file):
+        '''
+        Generate and write seeds in the CORSIKA input file.
+
+        Parameters
+        ----------
+        file: stream
+            File where the telescope positions will be written.
+        '''
+        randomSeed = self._userParameters['PRMPAR'][0] + self._userParameters['RUNNR'][0]
+        random.seed(randomSeed)
+        corsikaSeeds = [int(random.uniform(0, 1e7)) for i in range(4)]
```

Comment:
This needs further clarification. Seeds are generated with a random generator?

Reply:
The seed for the random module is calculated deterministically. And then the CORSIKA seeds are generated with the random module. This is reproducible.

Do you think we should set the seeds directly?

Reply:
For the prodN, we use the same run numbers several times (e.g., 180 deg and 0 deg files have the same run numbers, just a different MC az). Do I see it correctly that for the same runnumber, we get the same seeds in this case?

I would prefer to have a explicit random seed parameter - this make it clearer that we use a random generator, is easier to follow up, and does not depend on other parameters.

Reply:
Also note that the default random generator algorithm changed in python in the past, which completely messes up the idea of reproducibility even if you keep the seeds. 

I think we should somewhere make a note (an issue) to revisit how we want to use random generators in gammasim-tools. It is a central concept, and we might introduce a separate module where both seeds and algorithms are saved.

Reply:
> For the prodN, we use the same run numbers several times (e.g., 180 deg and 0 deg files have the same run numbers, just a different MC az). Do I see it correctly that for the same runnumber, we get the same seeds in this case?

Yes, but that is not a problem. We actually only need seeds that depends on the run number. If any other parameter is different (e.g. azimuth), we will not simulate the same showers.

The reproducibility issue goes beyond the python version. You can only reproduce the same showers if you use the same seeds and run CORSIKA in exactly the same system you ran it before. So using the same python version is only one of the requirements if one wants to simulate the same showers.

I created an issue about that.